include/nanobind/nb_attr.h

```diff
@@ -157,11 +157,7 @@ template <size_t Size> struct func_data_prelim {
     const char *doc;
     PyObject *scope;
 
-#if defined(_MSC_VER)
-    arg_data args[Size == 0 ? 1 : Size];
-#else
-    arg_data args[Size];
-#endif
+    arg_data args[Size == 0 ? 1 : Size];  // avoid zero-sized arrays
```

Comment:
This has an actual cost, for no practical purpose. I would prefer to silence the `-pendantic` warning from this line with a pragma or similar.